### DISCUSSION TITLE
Does this package has a "tail - f" (live) Modus?

### DISCUSSION THREAD

**Discussion by michabbb** [opcodesio/log-viewer#364] [Q&A]
To be clear, I am not talking about "update every x seconds" like here: https://github.com/opcodesio/log-viewer/discussions/289#discussion-5760235

A real live view would be nice, maybe with laravel reverb or something :)

**Comment by arukompas** [comment]
Hey @michabbb 

Would be cool! :) though I'm not quite sure whether it's worth introducing another dependency to the package.

I think for now simple polling every 2-5 seconds would be more than sufficient, easier to implement and not alienate existing package users with additional dependencies.

I think I had thought of introducing the auto-reload before, but I couldn't figure out how to nicely handle updates in the UI when new entries are detected. For example, you're on page 5, currently viewing log #45, and 2 seconds later there's 10 new entries. Does the page stay on page 5 and just load new entries for that page? Or does it stay on the same results but only change the pagination items? Perhaps auto-reload should only work when you're on the first page?

Pagination makes auto-reload a bit complex, but to refactor it to an infinite scrolling list would be too much work too.

So, that's my hesitation for now. Idea is great! But currently I don't have a good idea how to properly implement it so that everyone's happy.

**Reply by michabbb** [reply]
Pardon, wrong thinking. When watching a file in live mode there are no pages. That's a complete different concept and has nothing to do with pages. It's a constant stream of new stuff and it's get filled until you stop the live mode.

Don't know if the code is working, but here's an example:
Server 
```js
const fs = require('fs');
const http = require('http');
const path = require('path');

const filename = 'path/to/your/file.txt';
let file;

const server = http.createServer((req, res) => {
    if (req.url === '/stream') {
        
        if (!file) {
            file = fs.createReadStream(filename);
            file.on('data', (chunk) => {
                res.write(chunk);
            });
            file.on('end', () => {
                res.end();
            });
        }
        res.setHeader('Content-Type', 'text/event-stream');
        res.setHeader('Cache-Control', 'no-cache');
        res.setHeader('Connection', 'keep-alive');
    } else {
        
        res.writeHead(200, { 'Content-Type': 'text/html' });
        res.write(`
            <!DOCTYPE html>
            <html>
            <head>
                <title>Live File Watcher</title>
            </head>
            <body>
                <h1>Live File Watcher</h1>
                <pre id="file-content"></pre>
                <script src="client.js"></script>
            </body>
            </html>
        `);
        res.end();
    }
});

server.listen(8080, () => {
    console.log('Server running at http://localhost:8080/');
});
```
Client 
```js
const fileContentElement = document.getElementById('file-content');

const eventSource = new EventSource('/stream');

eventSource.addEventListener('message', (event) => {
    fileContentElement.textContent = event.data;
});
``;

**Reply by arukompas** [reply]
No, I understand that. 

What I'm saying is that currently the Log Viewer's frontend is built with pagination. So in order to have streamed log entries, we need to switch away from pagination (requires work), or figure out how to make streams work with pagination.

**Reply by michabbb** [reply]
Nope, no need to switch away. Don't know why you are so focused on that. The live view is just a page on its own, has nothing todo with your existing code (paging). But I don't want to start a discussion here, it's your code, do whatever you want. I just wanted to explain, that this would be a cool feature and it's easier to implement as you think. "tail -f" is totally fine for me, I just mentioned that because none-Linux users do benefit from that instead of pressing F5 100 times a day 😏

**Reply by arukompas** [reply]
Ah, lol, hadn't thought of it being a completely different page/view :) 

That's interesting, I'll note it down as a feature request, although these days I'm a little preoccupied with a newborn :) 

Great idea, thanks @michabbb ! 🙏
